sdk/evaluation/azure-ai-evaluation/tests/unittests/test_tool_call_accuracy_evaluator.py

```diff
@@ -396,3 +423,183 @@ def test_evaluate_tools_no_tools(self, mock_model_config):
+                    "output": "The financial data shows an unusual pattern where the total spending has significantly exceeded the total income in certain months.",
+                },
+            },
+        ]
+        tool_definitions = []
+        result = evaluator(query=query, tool_calls=tool_calls, tool_definitions=tool_definitions)
+
+        key = ToolCallAccuracyEvaluator._RESULT_KEY
+        assert result is not None
+        assert result[key] == 5.0
+        assert result[f"{key}_result"] == "pass"
+
+    def test_evaluate_code_interpreter(self, mock_model_config):
+        evaluator = ToolCallAccuracyEvaluator(model_config=mock_model_config)
+        evaluator._flow = MagicMock(side_effect=flow_side_effect)
+
+        # Test code interpreter for statistical analysis
+        query = "Find outliers in transaction amounts"
+        tool_calls = [
+            {
+                "type": "code_interpreter",
+                "code_interpreter": {
+                    "input": "import numpy as np\n\n# Calculate basic statistics\namounts = df['amount']\nmean = amounts.mean()\nstd = amounts.std()\n\n# Define outliers as transactions that are more than 2 standard deviations from the mean\noutlier_mask = (amounts < mean - 2*std) | (amounts > mean + 2*std)\noutliers = df[outlier_mask]\n\noutliers, mean, std",
+                    "outputs": [],
+                },
+            },
+        ]
+        tool_definitions = []
+        result = evaluator(query=query, tool_calls=tool_calls, tool_definitions=tool_definitions)
+
+        key = ToolCallAccuracyEvaluator._RESULT_KEY
+        assert result is not None
+        assert result[key] == 5.0
+        assert result[f"{key}_result"] == "pass"
+
+    def test_evaluate_sharepoint_grounding(self, mock_model_config):
+        evaluator = ToolCallAccuracyEvaluator(model_config=mock_model_config)
+        evaluator._flow = MagicMock(side_effect=flow_side_effect)
+
+        # Test SharePoint grounding for document search
+        query = "Find information about monthly saving rules"
+        tool_calls = [
+            {
+                "type": "sharepoint_grounding",
+                "sharepoint_grounding": {"input": "monthly saving rules", "output": "[redacted]"},
+            },
+        ]
+        tool_definitions = []
+        result = evaluator(query=query, tool_calls=tool_calls, tool_definitions=tool_definitions)
+
+        key = ToolCallAccuracyEvaluator._RESULT_KEY
+        assert result is not None
+        assert result[key] == 5.0
+        assert result[f"{key}_result"] == "pass"
+
+    def test_evaluate_open_api(self, mock_model_config):
+        evaluator = ToolCallAccuracyEvaluator(model_config=mock_model_config)
+        evaluator._flow = MagicMock(side_effect=flow_side_effect)
+
+        # Test OpenAPI function call for exchange rates
+        query = "What is the exchange rate from GBP to EUR?"
+        tool_calls = [
+            {
+                "type": "openapi",
```

Comment:
Does this work as expected? Does this tool call and also for other tools match tool calls as returned by converter ?